include/wil/Tracelogging.h

```diff
@@ -6597,15 +6602,15 @@ namespace details
 #else
             m_fireEventDelay = __TRACELOGGING_TEST_HOOK_API_TELEMETRY_EVENT_DELAY_MS;
 #endif
-            m_fireEventThreadPoolTimer.reset(::CreateThreadpoolTimer(
-                [](PTP_CALLBACK_INSTANCE, PVOID, PTP_TIMER) {
-                    FireEvent();
-                },
-                nullptr,
-                nullptr));
+            m_fireEventThreadPoolTimer.reset(::CreateThreadpoolTimer(&FireEventCallback, nullptr, nullptr));

```

Comment:
Based on this and the EnumWindows changes, I'm guessing that clang (or someone?) doesn't like the no-capture lambdas being passed as win32 API callbacks due to the calling convention?

I thought that that had been fixed as of https://github.com/llvm/llvm-project/commit/ec809e4cfe0be08b1bab7ba7e6ff905b9430571a, but I legitimately can't get it to work in Compiler Explorer...

Reply:
Oh okay, I did it in a really stupid way. You don't need to do this, but for discussion purposes:

https://godbolt.org/z/1q9YcYndr

```c++
extern "C" void CallbackWithStdcall(int(__stdcall * ParamType)());
extern "C" void CallbackWithCdecl(int(__cdecl * ParamType)());

extern "C" int GetExternalValue();

#ifndef _MSC_VER
#define LAMBDA_CALLING_CONVENTION __attribute__((stdcall))
#else
#define LAMBDA_CALLING_CONVENTION
#endif

void bah() {
    CallbackWithStdcall([]() LAMBDA_CALLING_CONVENTION { return GetExternalValue(); });
    CallbackWithCdecl([]{ return GetExternalValue(); });
}
```

this compiles "properly" under clang and MSVC with the target triple set appropriately

Reply:
That was the first thing I tried, however it caused the compiler to ICE in at least one circumstance...